src/lib/tasks/run-removal.ts

```diff
@@ -124,42 +116,48 @@ export function runRemoval(
         continue;
       }
 
-      // The sprite has been marked for removal, its in the right
+      // The sprite has been marked for removal, it's in the right
       // LifecyclePhase, and its time has expired. Flash zeros to the sprite's
       // data view and schedule it for a texture sync.
       properties.spriteView[DataViewSymbol].fill(0);
+      properties.zeroed = true;
       properties.lifecyclePhase = LifecyclePhase.NeedsTextureSync;
       coordinator.needsTextureSyncIndexRange.expandToInclude(properties.index);
+
+      // Break if we've run too long.
+      if (step++ % stepsBetweenChecks === 0 && remaining() <= 0) {
+        break;
+      }
     }
   } finally {
-    if (coordinator.needsTextureSyncIndexRange.isDefined) {
-      coordinator.queueTextureSync();
-    }
+    // Expand the toBeRemovedIndexRange and toBeRemovedTsRange to include
+    // any sprites that were not visited due to time.
+    for (let i = index + 1; i <= highIndex; i++) {
+      const sprite = coordinator.sprites[i];
+      const properties = sprite[InternalPropertiesSymbol];
 
-    if (index < highIndex) {
-      // Since we didn't finish the whole loop due to time, expand the index
-      // range to include all the indices which were previously marked, but
-      // which we didn't visit.
-      coordinator.toBeRemovedIndexRange.expandToInclude(index + 1);
-      coordinator.toBeRemovedIndexRange.expandToInclude(highIndex);
-
-      // Expand the Ts range to include the timestamps of the remaining sprites.
-      for (let i = index + 1; i <= highIndex; i++) {
-        const sprite = coordinator.sprites[i];
-        const properties = sprite[InternalPropertiesSymbol];
-        if (properties.toBeRemoved === true &&
-            properties.lifecyclePhase === LifecyclePhase.Rest) {
-          if (!properties.spriteView) {
-            // Indicates a bug in Megaplot. A Sprite in the Rest lifecycle phase
-            // ought to have been allocated a swatch and thus a SpriteView for
-            // interacting with it.
-            // eslint-disable-next-line no-unsafe-finally
-            throw new InternalError('Sprite lacks a SpriteView');
-          }
-          coordinator.toBeRemovedTsRange.expandToInclude(
-              properties.spriteView.TransitionTimeMs);
-        }
+      // Ignore any sprites that are not both in the Rest phase and have had
+      // their 'toBeRemoved' property set (had an exit callback).
+      if (!properties.toBeRemoved ||
+          properties.lifecyclePhase !== LifecyclePhase.Rest) {
+        continue;
+      }
+
+      if (!properties.spriteView) {
```

Comment:
nit: this is more of a personal style thing, but I have come to appreciate the specificity of `== null` checks rather than generic falsy evaluation, they help me distinguish between proper boolean value checks and existence checks.

Reply:
Switched to explicit `undefined` check.